### PR TITLE
Update test

### PR DESCRIPTION
**Pull request by samuel-massinon**
https://github.com/invenia/FTPClient.jl/pull/22 needs to be merged first.

### CODE REVIEW

File: src/FTPObject.jl [hunk 1]
```diff
@@ -1,35 +1,20 @@
...
-
-
 type FTP
     ctxt::ConnContext
 
-
-
-    function FTP(;host="", block=true, implt=false, ssl=false, ver_peer=true, act_mode=false, user="", pswd="", binary_mode=true)
-        options = RequestOptions(blocking=block, implicit=implt, ssl=ssl,
+    function FTP(;host="", implt=false, ssl=false, ver_peer=true, act_mode=false, user="", pswd="", binary_mode=true)
```
Comment (omus) [inline src/FTPObject.jl]: I know you didn't change the names of these keywords but some of these keywords could be renamed to be nicer:

`implt` -> `implicit`
`ver_peer` -> `verify`
`act_mode` -> `active`

File: src/FTPObject.jl [hunk 2]
```diff
@@ -1,35 +1,20 @@
...
+        options = RequestOptions(blocking=true, implicit=implt, ssl=ssl,
                     verify_peer=ver_peer, active_mode=act_mode,
                     username=user, passwd=pswd, hostname=host, binary_mode=binary_mode)
 
+        ctxt = nothing
         try
-            resp = ftp_connect(options)
+            ctxt, resp = ftp_connect(options)
         catch err
             if(isa(err, FTPClientError))
```
Comment (omus) [inline src/FTPObject.jl]: Reads cleaner as `if isa(err, FTPClientError)`

File: test/runtests.jl [hunk 3]
```diff
@@ -129,21 +169,23 @@ else
...
-        include(fp)
-    end
+        test_files = ["test_non_ssl.jl", "test_ftp_object.jl"]
 
-    # Basic commands will now error
-    set_errors()
+        for file in test_files
+            fp = joinpath(dirname(@__FILE__), file)
+            include(fp)
+        end
```
Comment (omus) [inline test/runtests.jl]: Isn't it simpler to just include the files directly here?

File: test/runtests.jl [hunk 4]
```diff
@@ -72,13 +112,15 @@ home_dir = "/"
...
 @windows_only byte_file_contents = string("466F6F426172", "0A", "466F6F426172", "1A1A1A")
 byte_file_name = randstring(20)
-upload_file = "test_upload.txt"
-f =  open(upload_file, "w")
-write(f, "Test file to upload.\n")
+upload_file_name = "test_upload.txt"
+f =  open(upload_file_name, "w")
+upload_file_contents = randstring(100)
+write(f, upload_file_contents)
 close(f)
```
Comment (omus) [inline test/runtests.jl]: The `open` do syntax should be used here

File: test/test_ftp_object.jl [hunk 5]
```diff
@@ -1,171 +1,325 @@
...
+    ftp_init()
+
+    @testset "with persistent connection" begin
+
+        function no_unexpected_changes(ftp::FTP)
+            @test ftp.ctxt.options.ssl == false
+            @test ftp.ctxt.options.username == user
+            @test ftp.ctxt.options.passwd == pswd
+            @test ftp.ctxt.options.hostname == host
+            @test ftp.ctxt.options.url == "ftp://"* host * "/"
```
Comment (omus) [inline test/test_ftp_object.jl]: String interpolation almost always looks cleaner: `"ftp://$host/"`

File: test/test_non_ssl.jl [hunk 6]
```diff
@@ -1,140 +1,291 @@
...
+        @test options.url == "ftp://" * string(host) * "/"
+
+        @testset "ftp_get" begin
+            resp = ftp_get(file_name, options)
+            actual_body = readstring(resp.body)
+            @test resp.code == 226
+            @test typeof(resp.total_time) == Float64
+            @test resp.bytes_recd == file_size == length(actual_body)
+            @test actual_body == file_contents
+            expected_header_last_part = AbstractString["200 TYPE completed.","502 Command not implemented: SIZE.","150 File status okay; about to open data connection.","226 Closing data connection. Requested file action successful."]
```
Comment (omus) [inline test/test_non_ssl.jl]: Having to cast this as an `AbstractString` array probably means something is wrong with the `Response` type

Comment (omus) [inline test/test_non_ssl.jl]: The [`Reponse` type](https://github.com/invenia/FTPClient.jl/blob/update_test/src/FTPC.jl#L36) should have a slight change:

``` julia
type Response{S<:AbstractString}
     ...
     headers::Array{S}
     ...
end
```

File: test/test_ftp_object.jl [hunk 7]
```diff
@@ -1,171 +1,325 @@
-using FTPClient
+@testset "FTPObject" begin
+
+    expected_header_port = r"229 Entering Extended Passive Mode \(\|\|\|\d*\|\)"
```
Comment (omus) [inline test/test_ftp_object.jl:3]: This regular expression is used all over the place. Consider making it a global constant.

File: test/test_ftp_object.jl [hunk 8]
```diff
@@ -1,171 +1,325 @@
...
-        @test resp.code == 226
-    end
-    @testset "uploading a file with remote local file name" begin
-        ftp = FTP(ssl=false, user=user, pswd=pswd, host=host)
-        open(upload_file) do local_file
-            resp = upload(ftp, local_file, "some other name")
+        @testset "readdir" begin
+            ftp = FTP(ssl=false, user=user, pswd=pswd, host=host)
+            dir = readdir(ftp)
+            @test setdiff(dir, ["test_directory",byte_file_name,"test_upload.txt","test_download.txt"]) == Array{ASCIIString,1}()
```
Comment (omus) [inline test/test_ftp_object.jl]: `Array{ASCIIString,1}()` is equal to `ASCIIString[]`

File: test/test_non_ssl.jl (hunk 6 above)
Comment (samuel-massinon) [inline test/test_non_ssl.jl]: What would be the benefits of changing the type? It seems like a more complicated constructor.

File: test/test_ftp_object.jl (hunk 7 above)
Comment (samuel-massinon) [inline test/test_ftp_object.jl:3]: I place all `expected_header_port` at the start of the files they are used in, so there's two of them. I could place it in runtests.jl, but I prefer to have the variables created and used in the same file for easy lookup. Thoughts.

**Comment by samuel-massinon** [comment]
@omus Ready to review.

File: test/test_ftp_object.jl (hunk 7 above)
Comment (omus) [inline test/test_ftp_object.jl:3]: I think it's ok to have it declared in runtests.jl and use it but I understand the concern. Once per file is a big improvement already so I am happy.

**Comment by omus** [comment]
I'm ok with merging.

**Pull request merged**